src/auth.rs

```diff
@@ -679,7 +679,7 @@ const DID_METHOD_WEB: &str = "web";
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub struct DidWeb {
-    domain: String,
+    domain: Arc<str>,
```

Comment:
What is the reason for wrapping the `domain` into `Arc`? 
I don't see any changes in how it's passed in the following code. Maybe I've missed it or it's for the further PRs.

Reply:
`Arc<str>` is much cheaper to clone than a `String` because it avoids allocating. I clone the domain a number of times. It follows similar patterns we have for ProjectId, Topic, AccountId, etc.